MonacoEditorComponent/ts-helpermethods/updateSelectedContent.ts

```diff
@@ -0,0 +1,48 @@
+﻿///<reference path="../monaco-editor/monaco.d.ts" />
+declare var Parent: ParentAccessor;
+declare var Keyboard: KeyboardListener;
+
+declare var editor: monaco.editor.IStandaloneCodeEditor;
+declare var model: monaco.editor.ITextModel;
+declare var contexts: { [index: string]: monaco.editor.IContextKey<any> };//{};
+declare var decorations: string[];
+declare var modifingSelection: boolean; // Supress updates to selection when making edits.
+
+var updateSelectedContent = function (content) {
+    let selection = editor.getSelection();
+
+    // Need to ignore updates from us notifying of a change
+    if (content != model.getValueInRange(selection)) {
+        modifingSelection = true;
+        let range = new monaco.Range(selection.startLineNumber, selection.startColumn, selection.endLineNumber, selection.endColumn);
+        let op = { identifier: { major: 1, minor: 1 }, range, text: content, forceMoveMarkers: true };
+
+        // Make change to selection
+        //TODO how to properly fix this code?
```

Comment:
Please review these TODOs

Reply:
Could you clarify what the current issue with it is?  I'll pull down the PR later to test it out, but just doing a quick pass at the moment before dinner.  What should I be looking for?

Reply:
There were typescript compile errors, I tried to fix them but I didn’t know what the intended behavior is.

I commented out the parts with compile errors and replaced them with “something that looks similar” but they need a more knowledgeable eye to look at them I believe.

Reply:
Ah, I understand now, I'll see if I can take a look later this week sometime.

Reply:
@hawkerm have you been able to look at it? Would be nice if I could improve the PR with your feedback :)